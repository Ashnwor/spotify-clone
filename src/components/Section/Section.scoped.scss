@import '../../styles/color-palette';

.section {
  display: flex;
  height: 300px;
  flex-direction: column;
  background: $color-black-3;

  @media (max-width: 768px) {
    height: 220px;
  }
}

.titlebar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-bottom: 12px;
}

.title {
  height: 33px;
  color: $color-white-1;
  font-family: 'Segoe UI';
  font-size: 24px;
  font-style: normal;
  font-weight: 700;
  line-height: 28px;
  text-overflow: ellipsis;
  white-space: nowrap;

  @media (max-width: 768px) {
    font-size: 20px;
  }
}

.titlebar-button {
  background-color: transparent;
  color: $color-gray-3;
  cursor: pointer;
  font-family: 'Segoe UI';
  font-size: 12px;
  font-style: normal;
  font-weight: 700;
  line-height: 16px;

  &:hover {
    text-decoration: underline;
  }
}

.content {
  display: grid;
  width: 100%;
  height: 251px;
  grid-template-columns: repeat(auto-fill, minmax(164px, 1fr));
  background: $color-black-3;
  column-gap: 16px;
  overflow-y: hidden;

  @media (max-width: 768px) {
    height: 200px;
    grid-template-columns: repeat(auto-fill, 124px);
    column-gap: 8px;
  }
}
