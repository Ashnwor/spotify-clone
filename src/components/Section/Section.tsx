import React from 'react';
import './Section.scoped.scss';
import Card from '../Card/Card';
import Song from '../../types/Song';

interface Props {
  title: string;
  data: Song[];
}

const Section: React.FC<Props> = ({ title, data }) => {
  return (
    <div className="section">
      <div className="titlebar">
        <div className="title">{title}</div>
        <div className="titlebar-button">SEE ALL</div>
      </div>
      <div className="content">
        {data.map((song: Song) => {
          return <Card song={song} key={song._id} />;
        })}
      </div>
    </div>
  );
};

export default Section;
