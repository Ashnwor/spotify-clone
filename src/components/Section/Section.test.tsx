import { shallow } from 'enzyme';
import Section from './Section';

it('runs correctly', () => {
  const wrapper = shallow(
    <Section
      data={[
        { _id: 'test_id_1', name: 'test_name', cover: '', artist: 'test_artist' },
        { _id: 'test_id_2', name: 'test_name', cover: '', artist: 'test_artist' },
        { _id: 'test_id_3', name: 'test_name', cover: '', artist: 'test_artist' },
      ]}
      title="test"
    />,
  );
  expect(wrapper.find('.content').children()).toHaveLength(3);
  expect(wrapper).toMatchSnapshot();
});
