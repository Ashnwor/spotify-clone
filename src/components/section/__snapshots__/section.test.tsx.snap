// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs correctly 1`] = `
<div
  className="section"
>
  <div
    className="section-titlebar"
  >
    <div
      className="section-title"
    >
      <span
        className="section-title-text"
      >
        test
      </span>
    </div>
    <div
      className="section-titlebar-btn"
    >
      <span
        className="section-titlebar-btn-text"
      >
        SEE ALL
      </span>
    </div>
  </div>
  <div
    className="section-content"
  >
    <Card
      key="test_id_1"
      song={
        Object {
          "_id": "test_id_1",
          "artist": "test_artist",
          "cover": "",
          "name": "test_name",
        }
      }
    />
    <Card
      key="test_id_2"
      song={
        Object {
          "_id": "test_id_2",
          "artist": "test_artist",
          "cover": "",
          "name": "test_name",
        }
      }
    />
    <Card
      key="test_id_3"
      song={
        Object {
          "_id": "test_id_3",
          "artist": "test_artist",
          "cover": "",
          "name": "test_name",
        }
      }
    />
  </div>
</div>
`;
